### ISSUE TITLE
[doc] being able to switch version while keeping the same doc

### ISSUE THREAD

**Issue by stas00** [pytorch/pytorch#51575]
## 📚 Documentation

Often I go to a specific document, say, https://pytorch.org/docs/stable/optim.html and then I realize that it's not `master`, so I want to read the bleed-edge doc of the same, so I go to the left upper corner and switch to `master`, but it then takes me to the front page, which is not what I want and I have to search for the page I want again. I want it to take me to https://pytorch.org/docs/master/optim.html. 

This is a feature request to enable dynamic switching of the version while keeping the opened page the same (just different version).

And yes, I can hack the url to replace `stable` with `master`, but there must be an easier way to do it.

Thank you!

**Comment by mruberry** [comment]
That's a cool idea. cc @brianjo

**Comment by brianjo** [comment]
I like that idea. We're working on some different navigation options to move between versions, so I'll try to add that to our feature if I can.

**Comment by stas00** [comment]
any updates on this UI?

Since torch doesn't say when a certain function was added, it's very difficult to use in frameworks that have to support older pytorch versions.

I had to look for when `torch.linalg.norm` was added to properly support older pytorch in HF Transformers, and it's very slow and painful process.

I could have looked for it using the doc version switching version and I find out the version where it disappears, e.g. going from 

https://pytorch.org/docs/stable/generated/torch.linalg.norm.html
https://pytorch.org/docs/1.9.0/generated/torch.linalg.norm.html
https://pytorch.org/docs/1.8.0/generated/torch.linalg.norm.html

via switching the version number in the left menu. But this is not possible, it takes me to the main page of the release :(

Thank you.

**Comment by mruberry** [comment]
You're right, @stas00, we should also start listing the first version an operation is available from in the documentation -- sorry that's been a pain
